### PR TITLE
[IMP] decimal precision rounding in balance diffs, on payment entry

### PR DESCRIPTION
**Pull request by archetipo**
```
  adding max_balance_diff field in company configuration
```

### CODE REVIEW

**Comment by eLBati** [comment]
LGTM :+1: 

Travis error seems not related.
It also occurs in 8.0 branch https://travis-ci.org/OCA/account-payment/branches
However, I can't understand why the error occurs. I am not able to reproduce it locally

File: account_voucher_cash_basis/company.py [hunk 1]
```diff
@@ -32,6 +32,15 @@ class ResCompany(orm.Model):
             help="If not set, paying several 'cash basis' invoices with same "
                  "voucher with write-off won't be allowed. If set, write-off "
                  "will be distributed equally over invoices"),
+        'max_balance_diff': fields.integer(
+            string="Maximum unbalanced of payment entry (cents)",
```
Comment (eLBati) [inline account_voucher_cash_basis/company.py]: @archetipo 

> Max balancing precision in cash basis

File: account_voucher_cash_basis/company.py [hunk 2]
```diff
@@ -45,6 +54,13 @@ class AccountConfigSettings(orm.TransientModel):
             help="If not set, paying several 'cash basis' invoices with same "
                  "voucher with write-off won't be allowed. If set, write-off "
                  "will be distributed equally over invoices"),
+        'max_balance_diff': fields.related(
+            'company_id', 'max_balance_diff',
+            type="integer",
+            string="Maximum unbalanced of payment entry (cents)",
```
Comment (eLBati) [inline account_voucher_cash_basis/company.py]: @archetipo 

> Max balancing precision in cash basis

File: account_voucher_cash_basis/company.py (hunk 1 above)
Comment (archetipo) [inline account_voucher_cash_basis/company.py]: thanks

File: account_voucher_cash_basis/company.py [hunk 3]
```diff
@@ -32,6 +32,15 @@ class ResCompany(orm.Model):
             help="If not set, paying several 'cash basis' invoices with same "
                  "voucher with write-off won't be allowed. If set, write-off "
                  "will be distributed equally over invoices"),
+        'max_balance_diff': fields.integer(
+            string="Maximum unbalanced of payment entry (cents)",
+            help="This field is used in payment entry"
+                 " to check if move lines are ,"
+                 " unbalanced for more than this value."
```
Comment (eLBati) [inline account_voucher_cash_basis/company.py]: @archetipo 

> In cash basis payments, while balancing unbalanced entries, this is maximum error allowed

File: account_voucher_cash_basis/company.py [hunk 4]
```diff
@@ -45,6 +54,13 @@ class AccountConfigSettings(orm.TransientModel):
             help="If not set, paying several 'cash basis' invoices with same "
                  "voucher with write-off won't be allowed. If set, write-off "
                  "will be distributed equally over invoices"),
+        'max_balance_diff': fields.related(
+            'company_id', 'max_balance_diff',
+            type="integer",
+            string="Maximum unbalanced of payment entry (cents)",
+            help="This field is used in payment entry"
+                 " to check if move lines are ,"
+                 " unbalanced for more than this value."),
```
Comment (eLBati) [inline account_voucher_cash_basis/company.py]: @archetipo 

> In cash basis payments, while balancing unbalanced entries, this is maximum error allowed

**Comment by archetipo** [comment]
@eLBati thanks done

**Comment by eLBati** [comment]
Thanks :+1:

**Comment by pedrobaeza** [comment]
Travis error unrelated, so I merge.

**Pull request merged**